app/script/calling/entities/ECall.coffee

```diff
@@ -0,0 +1,364 @@
+      amplify.publish z.event.WebApp.AUDIO.STOP, z.audio.AudioType.NETWORK_INTERRUPTION
+
+    @participants_count.subscribe (users_in_call) =>
+      @max_number_of_participants = Math.max users_in_call, @max_number_of_participants
+
+    @self_client_joined.subscribe (is_joined) =>
+      return if is_joined
+      @is_connected false
+      amplify.publish z.event.WebApp.AUDIO.PLAY, z.audio.AudioType.CALL_DROP if @state() in [z.calling.enum.CallState.DISCONNECTING, z.calling.enum.CallState.ONGOING]
+      @telemetry.track_duration @
+      @_reset_timer()
+      @_reset_e_flows()
+
+    @state.subscribe (state) =>
+      @logger.debug "E-call state '#{@id}' changed to '#{state}'"
+
+      @_clear_state_timeout()
+
+      if state in z.calling.enum.CallStateGroups.STOP_RINGING
+        @_on_state_stop_ringing()
+      else if state in z.calling.enum.CallStateGroups.IS_RINGING
+        @_on_state_start_ringing state is z.calling.enum.CallState.INCOMING
+
+      if state is z.calling.enum.CallState.CONNECTING
+        attributes = direction: if @previous_state is z.calling.enum.CallState.OUTGOING then z.calling.enum.CallState.OUTGOING else z.calling.enum.CallState.INCOMING
+        @telemetry.track_event z.tracking.EventName.CALLING.JOINED_CALL, @, attributes
+
+      @previous_state = state
+
+    @conversation_et.call @
+
+  update_timer_duration: =>
+    @duration_time Math.floor (Date.now() - @timer_start) / 1000
+
+
+  ###############################################################################
+  # Call states
+  ###############################################################################
+
+  send_e_call_event: (e_call_message_et) =>
+    @e_call_center.send_e_call_event @conversation_et, e_call_message_et
+
+  start_negotiation: =>
+    @self_client_joined true
+    @self_user_joined true
+    e_participant_et.e_flow_et.start_negotiation() for e_participant_et in @participants() when e_participant_et.e_flow_et
+
+  _on_state_start_ringing: (is_incoming) =>
+    @_play_call_sound is_incoming
+    @_set_state_timeout is_incoming
+
+  _on_state_stop_ringing: =>
+    if @previous_state in z.calling.enum.CallStateGroups.IS_RINGING
+      @_stop_call_sound @previous_state is z.calling.enum.CallState.INCOMING
+
+  _clear_state_timeout: =>
+    window.clearTimeout @state_timeout
+    @state_timeout = undefined
+
+  _play_call_sound: (is_incoming) ->
+    sound_id = if is_incoming then z.audio.AudioType.INCOMING_CALL else z.audio.AudioType.OUTGOING_CALL
+    amplify.publish z.event.WebApp.AUDIO.PLAY_IN_LOOP, sound_id
+
+  _set_state_timeout: (is_incoming) =>
+    @state_timeout = window.setTimeout =>
+      @_stop_call_sound is_incoming
+      if is_incoming
+        return @state z.calling.enum.CallState.IGNORED if @is_group()
+        amplify.publish z.event.WebApp.CALL.STATE.DELETE, @id
+      else
+        amplify.publish z.event.WebApp.CALL.STATE.LEAVE, @id
+    , 30000
```

Comment:
Should `30000` become a config value? Like `CALLING_CONFIG.CONFIG_UPDATE_INTERVAL`?